src/fourier_resizing.jl

```diff
@@ -138,60 +138,85 @@ function rft_pad(mat, new_size)
     return select_region(mat;new_size=new_size, center=c2.+1)
 end
 
-function ft_fix_before(mat, size_old, size_new; start_dim=1)
-    for d = start_dim:ndims(mat)
-        sn = size_new[d]
-        so = size_old[d]
-        if sn < so && iseven(sn)
-            L1 = (size_old[d] -size_new[d] )÷2 +1
-            mat = FourierJoin(mat, d, L1)
-        end
+"""
+    ft_fix_before(mat::MT, size_old, size_new, ::Val{N})::FourierJoin{T,N,MT}  where {T,N, MT<:AbstractArray{T,N}} 
+    implements the specialized (highest dimension) version of a recursive dimension-specific function that returns an array type which knows
```

Comment:
empty line after function definition

and move the to the very left:

So the text should not be tabbed to the right

Reply:
done